src/transformers/trainer.py

```diff
@@ -3741,6 +3751,7 @@ def evaluate(
         eval_dataset: Optional[Union[Dataset, Dict[str, Dataset]]] = None,
         ignore_keys: Optional[List[str]] = None,
         metric_key_prefix: str = "eval",
+        **gen_kwargs,
```

Comment:
I think it is better if we don't introduce this as it is redundant. People who want to modify their `generation_config` can just pass in it `TrainingArguments`. I think in the seq2seq trainer, `gen_kwargs` was added first, then came `generation_config`. If you are okay, we can simplify the `predict` and `predict_step` method. cc @gante